tests/unit/queries/attribute-test.js

```diff
@@ -55,3 +55,11 @@ it('uses page scope', function(assert) {
 
   assert.equal(attr(), 'Logo small');
 });
+
+it('searches for element by index if provided', function(assert) {
+  fixture('<img alt="img1" class="img"></img><img alt="img2" class="img"></img>');
```

Comment:
I'm not sure if I follow, you don't have to use nested child elements to use the index options, its just a way to select from any matched elements by a certain selector.